Marlin/src/inc/SanityCheck.h

```diff
@@ -4207,3 +4202,33 @@ static_assert(_PLUS_TEST(4), "HOMING_FEEDRATE_MM_M values must be positive.");
 #if ENABLED(XFER_BUILD) && !BOTH(BINARY_FILE_TRANSFER, CUSTOM_FIRMWARE_UPLOAD)
   #error "BINARY_FILE_TRANSFER and CUSTOM_FIRMWARE_UPLOAD are required for custom upload."
 #endif
+
+// Check requirements for Input Shaping
+#if ENABLED(INPUT_SHAPING) && defined(__AVR__)
+  #if HAS_SHAPING_X && (SHAPING_FREQ_X) * 2 * 0x10000 < (STEPPER_TIMER_RATE)
+    #if F_CPU > 16000000
+      #error "SHAPING_FREQ_X is below the minimum (20) for AVR 20MHz."
+    #else
+      #error "SHAPING_FREQ_X is below the minimum (16) for AVR 16MHz."
+    #endif
+  #elif HAS_SHAPING_Y && (SHAPING_FREQ_Y) * 2 * 0x10000 < (STEPPER_TIMER_RATE)
+    #if F_CPU > 16000000
+      #error "SHAPING_FREQ_Y is below the minimum (20) for AVR 20MHz."
+    #else
+      #error "SHAPING_FREQ_Y is below the minimum (16) for AVR 16MHz."
+    #endif
+  #endif
+#endif
+
+#if ENABLED(INPUT_SHAPING)
+  #if ENABLED(DIRECT_STEPPING)
+    #error "INPUT_SHAPING cannot currently be used with DIRECT_STEPPING."
```

Comment:
Theoretically these should be able to coexist, with either the direct stepping blocks being shaped or by having the direct stepping blocks skip the shaper altogether.

Reply:
Agreed. They may even work okay as is. I think direct stepping would probably just not be affected by input shaping. But I don't have the capacity to check that out right now.

Reply:
I assume that Klipper's direct stepping protocol sends basic segments but the shaping happens on the printer's mainboard, so it makes sense to also do that for Direct Stepping. But maybe Klipper does some pre-calculation for the benefit of shaping before it sends those segments over.

Reply:
Do we use the same block protocol as Klipper? Or is there a front end out there somewhere for feeding Marlin specifically?

Reply:
Direct Stepping G6 specifically requires https://github.com/colinrgodsey/step-daemon